modin/test/exchange/dataframe_protocol/omnisci/test_protocol.py

```diff
@@ -0,0 +1,266 @@
+# Licensed to Modin Development Team under one or more contributor license agreements.
+# See the NOTICE file distributed with this work for additional information regarding
+# copyright ownership.  The Modin Development Team licenses this file to you under the
+# Apache License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+
+"""Dataframe exchange protocol tests that are specific for OmniSci implementation."""
+
+import pytest
+import pyarrow as pa
+import pandas
+import numpy as np
+
+import modin.pandas as pd
+from modin.core.dataframe.pandas.exchange.dataframe_protocol.from_dataframe import (
+    convert_primitive_column_to_ndarray,
+    buffer_to_ndarray,
+    set_nulls,
+)
+from modin.pandas.utils import from_arrow, from_dataframe
+from modin.pandas.test.utils import df_equals
+from modin.test.test_utils import warns_that_defaulting_to_pandas
+from .utils import get_data_of_all_types, split_df_into_chunks, export_frame
+
+
+@pytest.mark.parametrize("data_has_nulls", [True, False])
+@pytest.mark.parametrize("from_omnisci", [True, False])
+def test_simple_export(data_has_nulls, from_omnisci):
+    if from_omnisci:
+        # OmniSci can't import 'uint64' as well as booleans
+        # issue for bool: https://github.com/modin-project/modin/issues/4299
+        exclude_dtypes = ["bool", "uint64"]
+    else:
+        exclude_dtypes = None
+
+    data = get_data_of_all_types(
+        has_nulls=data_has_nulls, exclude_dtypes=exclude_dtypes
+    )
+    md_df = pd.DataFrame(data)
+
+    exported_df = export_frame(md_df, from_omnisci)
+    df_equals(md_df, exported_df)
+
+    exported_df = export_frame(md_df, from_omnisci, nchunks=3)
```

Comment:
Let's make n_chunks as a parameter of this test.

Reply:
What is the original number of chunks for md_df? Does it very depend on the data generated?

Reply:
> Let's make n_chunks as a parameter of this test.

Done

> What is the original number of chunks for md_df? Does it very depend on the data generated?

The original number of chunks is always equal to 1 (the PyArrow table is always unchunked), it does not depend on the data.